File: programas_python/911/[#11].py
'''[#11] Exiba uma lista de opções (menu): adição, subtração, divisão, multiplicação e sair. Imprima a 
tabuada da operação escolhida. Repita até a opção saída seja escolhida.'''

def main():
    while True:
        print("\nEscolha uma operação:")
        print("1 - Adição")
        print("2 - Subtração")
        print("3 - Multiplicação")
        print("4 - Divisão")
        print("5 - Sair")
        
        opcao = int(input("Digite sua opção: "))
        
        if opcao == 5:
            break
        
        if opcao >= 1 and opcao <= 4:
            numero = float(input("\nDigite um número para a tabuada: "))

            for i in range(1, 11):
                if opcao == 1:
                    print(f"{numero} + {i} = {numero + i:.0f}")
                elif opcao == 2:
                    print(f"{numero} - {i} = {numero - i:.0f}")
                elif opcao == 3:
                    print(f"{numero} * {i} = {numero * i:.0f}")
                elif opcao == 4:
                    if i != 0:
                        print(f"{numero} / {i} = {numero / i:.2f}")
                    else:
                        print("Divisão por zero não é permitida.")
        else:
            print("\nERR.\n\n")

if __name__ == "__main__":
    main()

File: programas_C/[#11].c
/*[#11] Exiba uma lista de opções (menu): adição, subtração, divisão, multiplicação e sair. Imprima a 
tabuada da operação escolhida. Repita até a opção saída seja escolhida.*/

#include <stdio.h>
int main() {
    int opcao;
    float numero;

    do {
        printf("Escolha uma operação:\n");
        printf("1 - Adição\n");
        printf("2 - Subtração\n");
        printf("3 - Multiplicação\n");
        printf("4 - Divisão\n");
        printf("5 - Sair\n");
        printf("Digite sua opção: ");
        scanf("%d", &opcao);
    
        if (opcao == 5) {
            break;
        }

        if (opcao >= 1 && opcao <= 4) {
            printf("\nDigite um número para a tabuada: ");
            scanf("%f", &numero);

            for (int i = 1; i <= 10; i++) {
                switch (opcao) {
                    case 1:
                        printf("%.0f + %d = %.0f\n", numero, i, numero + i);
                        break;
                    case 2:
                        printf("%.0f - %d = %.0f\n", numero, i, numero - i);
                        break;
                    case 3:
                        printf("%.0f * %d = %.0f\n", numero, i, numero * i);
                        break;
                    case 4:
                        if (i != 0) {
                            printf("%.0f / %d = %.2f\n", numero, i, numero / i);
                        } else {
                            printf("Divisão por zero não permitida.\n");
                        }
                        break;

                    default:
                        printf("Operação inválida.\n");
                        break;
                }
            }
        } else {
            printf("\nERR.\n\n");}
        
    } while (opcao != 5);

    return 0;
}